src/internal/required_argument.py

```diff
@@ -0,0 +1,18 @@
+from internal.argument import Argument
+
+# Class to represent a required argument to a command line tool
+class RequiredArgument(object):
+    def __init__(self, argument=None):
+        self.argument = argument
+
+    # Function to build a RequiredArgument object based on collections objects
+    # Raises KeyError if the argument's name isn't present
+    @classmethod
+    def build(cls, argument_data):
```

Comment:
This function doesn't quite work as it should. The object passed into this function as `argument_data` is the dictionary containing `name`, `type`, and `values` from `cli.yml`. This function should pass all of them to `Argument` and should probably throw an exception if the `name` field doesn't get set (all required arguments should be defined with a name).